packages/elements/src/react/common/form/index.tsx

```diff
@@ -225,91 +267,139 @@ function FieldState({ children }: { children: (state: { state: FieldStates }) =>
   return children(fieldState);
 }
 
-type FormInputProps = FormControlProps | ({ type: 'otp' } & OTPInputProps);
+/* -------------------------------------------------------------------------------------------------
+ * Input
+ * -----------------------------------------------------------------------------------------------*/
 
-function Input(props: FormInputProps) {
-  const field = useInput(props);
+const INPUT_NAME = 'ClerkElementsLabel';
+
+type FormInputProps = RadixFormControlProps | ({ type: 'otp' } & OTPInputProps);
 
+const Input = (props: FormInputProps) => {
+  const field = useInput(props);
   return <field.Element {...field.props} />;
-}
+};
 
-function Label(props: FormLabelProps) {
-  return <RadixLabel {...props} />;
-}
+Input.displayName = INPUT_NAME;
 
-type FormSubmitProps = SetRequired<RadixFormSubmitProps, 'children'>;
+/* -------------------------------------------------------------------------------------------------
+ * Label
+ * -----------------------------------------------------------------------------------------------*/
 
-function Submit(props: FormSubmitProps) {
-  return <RadixSubmit {...props} />;
-}
+const LABEL_NAME = 'ClerkElementsLabel';
 
-// ================= ERRORS ================= //
+const Label = RadixLabel;
 
-type FormErrorRenderProps = Pick<ClerkElementsError, 'code' | 'message'>;
-type FormErrorProps<T> = Omit<T, 'asChild' | 'children'> &
-  (
-    | {
-        children?: (error: FormErrorRenderProps) => React.ReactNode;
-        code?: string;
-      }
-    | {
-        children: React.ReactNode;
-        code: string;
-      }
-  );
+Label.displayName = LABEL_NAME;
 
-type FormGlobalErrorProps = FormErrorProps<ComponentProps<'div'>>;
-type FormFieldErrorProps = FormErrorProps<FormMessageProps & { name?: string }>;
+/* -------------------------------------------------------------------------------------------------
+ * Submit
+ * -----------------------------------------------------------------------------------------------*/
 
-function GlobalError({ children, code, ...rest }: FormGlobalErrorProps) {
-  const { errors } = useGlobalErrors();
+const SUBMIT_NAME = 'ClerkElementsSubmit';
 
-  const error = errors?.[0];
+type FormSubmitProps = SetRequired<RadixFormSubmitProps, 'children'>;
+type FormSubmitComponent = React.ForwardRefExoticComponent<FormSubmitProps & React.RefAttributes<HTMLButtonElement>>;
 
-  if (!error || (code && error.code !== code)) {
-    return null;
-  }
+const Submit = RadixSubmit as FormSubmitComponent;
```

Comment:
`FormSubmitComponent` is exactly the same as the original component except that the `children` is required. Using `as` allows us to accomplish this without the (albeit minimal) runtime tax we paid previously.